dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -134,15 +136,20 @@ static void stripContent(List<? extends Entry> entries, AbstractDockstoreDAO dao
             dao.evict(entry);
             // clear users which are also lazy loaded
             entry.setUsers(null);
-            // need to have this evicted so that hibernate does not actually delete the tags and users
-            Set<Version> versions = entry.getWorkflowVersions();
-            versions.forEach(version ->
-                version.getSourceFiles().forEach(sourceFile ->
-                        ((SourceFile)sourceFile).setContent(null))
-            );
         }
     }
 
+    static void removeSourceFilesFromEntry(Entry entry, SessionFactory sessionFactory) {
```

Comment:
Can you add a comment to the method explaining what is going on here? If I knew Hibernate better, I might understand. But I don't. I sort of see what's happening, but I don't think I grok all the details.

Reply:
Added just the comment for now. Tried doing just the first suggestion, but ran into some issues when implementing 
the new getFileDAO function in this file https://github.com/dockstore/dockstore/blob/feature/3712/lazyLoadSourcefiles/dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java
Doesn't seem simple to fix after talking to Charles.

@denis-yuen do you have comment on whether these suggestions should be done?

Reply:
I don't have a preference as to whether the query should be in EntryDAO, but not really a hibernate expert. 
(I think there can be whole careers in this space)

I agree that it would be nice to avoid passing FileDAO everywhere but I don't feel strongly about it if it seems very problematic.

Reply:
Sorry, probably less enlightening than hoped